library/src/main/java/com/novoda/downloadmanager/File.java

```diff
@@ -0,0 +1,137 @@
+
+        File file = (File) o;
+
+        if (networkAddress != null ? !networkAddress.equals(file.networkAddress) : file.networkAddress != null) {
+            return false;
+        }
+        if (downloadFileId != null ? !downloadFileId.equals(file.downloadFileId) : file.downloadFileId != null) {
+            return false;
+        }
+        if (fileName != null ? !fileName.equals(file.fileName) : file.fileName != null) {
+            return false;
+        }
+        return relativePath != null ? relativePath.equals(file.relativePath) : file.relativePath == null;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = networkAddress != null ? networkAddress.hashCode() : 0;
+        result = 31 * result + (downloadFileId != null ? downloadFileId.hashCode() : 0);
+        result = 31 * result + (fileName != null ? fileName.hashCode() : 0);
+        result = 31 * result + (relativePath != null ? relativePath.hashCode() : 0);
+        return result;
+    }
+
+    @Override
+    public String toString() {
+        return "File{"
+                + "networkAddress='" + networkAddress + '\''
+                + ", downloadFileId=" + downloadFileId
+                + ", fileName=" + fileName
+                + ", relativePath=" + relativePath
+                + '}';
+    }
+
+    public interface Builder {
+        Builder withDownloadFileId(DownloadFileId downloadFileId);
+
+        Builder withFileName(FileName fileName);
+
+        Builder withRelativePath(String relativePath);
+
+        Batch.Builder apply();
+    }
+
+    interface InternalBuilder extends Builder {
+        Builder withParentBuilder(Batch.InternalBuilder parentBuilder);
+    }
+
+    private static final class LiteFileBuilder implements InternalBuilder {
+
+        private final String networkAddress;
+        private Optional<DownloadFileId> downloadFileId = Optional.absent();
+        private Optional<FileName> fileName = Optional.absent();
+        private Optional<String> relativePath = Optional.absent();
+
+        private Batch.InternalBuilder parentBuilder;
+
+        LiteFileBuilder(String networkAddress) {
+            this.networkAddress = networkAddress;
+        }
+
+        @Override
+        public Builder withParentBuilder(Batch.InternalBuilder parentBuilder) {
+            this.parentBuilder = parentBuilder;
+            return this;
+        }
+
+        @Override
+        public Builder withDownloadFileId(DownloadFileId downloadFileId) {
+            this.downloadFileId = Optional.of(downloadFileId);
+            return this;
+        }
+
+        @Override
+        public Builder withFileName(FileName fileName) {
+            this.fileName = Optional.of(fileName);
+            return this;
+        }
+
+        @Override
+        public Builder withRelativePath(String relativePath) {
+            this.relativePath = Optional.of(relativePath);
+            return this;
+        }
+
+        @Override
+        public Batch.Builder apply() {
```

Comment:
What's the thinking behind this method?

Reply:
So you can return the parent builder so you have a fluent builder, one where you can continue to add files.